packages/lesswrong/lib/helpers.ts

```diff
@@ -32,26 +33,43 @@ export function constantTimeCompare({ correctValue, unknownValue }: { correctVal
   }
 }
 
-// LESSWRONG version of getting unused slug. Modified to also include "oldSlugs" array
+// Get an unused slug. If `slug` is already unused, returns it as-is. If it's
+// used, finds a suffix such that slug-{suffix} is unused.
+//
+// Slugs are sequential up to 10, then jump to 4-char IDs, then 8-char IDs.
+//
+// If useOldSlugs is true, then the slugs that renamed documents used to have
+// count as used. If a documentId is provided, that document doesn't count as
+// a collision.
 Utils.getUnusedSlug = async function <T extends HasSlugType>(collection: CollectionBase<HasSlugType>, slug: string, useOldSlugs = false, documentId?: string): Promise<string> {
   let suffix = '';
   let index = 0;
   
-  let existingDocuments = await getDocumentsBySlug({slug, suffix, useOldSlugs, collection})
-  // test if slug is already in use
-  while (!!existingDocuments?.length) {
+  //eslint-disable-next-line no-constant-condition
+  while(true) {
```

Comment:
wow, this is scary

Reply:
This is already in master; it probably shouldn't be in the diff.

(It's structured this way because it's not otherwise possible to avoid duplicating part of the loop for the first iteration)